test/extended/machine_config/pinnedimages.go

```diff
@@ -0,0 +1,241 @@
+package machine_config
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+
+	mcfgv1alpha1 "github.com/openshift/api/machineconfiguration/v1alpha1"
+	mcClient "github.com/openshift/client-go/machineconfiguration/clientset/versioned"
+	exutil "github.com/openshift/origin/test/extended/util"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/yaml"
+)
+
+// This test is [Serial] because it modifies the state of the images present on Node in each test.
+var _ = g.Describe("[sig-mco][OCPFeatureGate:PinnedImages][OCPFeatureGate:MachineConfigNodes][Serial]", func() {
+	defer g.GinkgoRecover()
+	var (
+		MCOPinnedImageBaseDir            = exutil.FixturePath("testdata", "machine_config", "pinnedimage")
+		pinnedImageSetFixture            = filepath.Join(MCOPinnedImageBaseDir, "pis.yaml")
+		customMCP                        = filepath.Join(MCOPinnedImageBaseDir, "customMCP.yaml")
```

Comment:
Non-blocking nit-picky suggestion.
```suggestion
		customMCPFixture                        = filepath.Join(MCOPinnedImageBaseDir, "customMCP.yaml")
```

Reply:
agreed. updated